base/shared/src/main/scala/scalaz/typeclass/Debug.scala

```diff
@@ -1,19 +1,28 @@
 package scalaz
 package typeclass
 
-/** A typeclass describing types which can be meaningfully represented as a `String`.
+import scalaz.data.Cord
+
+/** A typeclass describing types which can be meaningfully represented as a `String`, for debugging purposes.
   */
 trait Debug[A] {
   /** Produce a `String` representation of `a`.
     */
-  /* todo: when we have a `Cord`, change this to return that instead,
-   * todo: and add a `def debugs(a: A): String = debug(a).toString`
-   */
-  def debug(a: A): String
+  def debug(a: A): Cord = Cord(debugs(a))
+  def debugs(a: A): String = Cord.fold(debug(a))
```

Comment:
At least on of the `debug[s]` methods must be overridden: would be safer to use the `Alt` trait pattern and let the two methods abstract.

Reply:
Gotcha.